pilot/pkg/serviceregistry/kube/controller/ambient/cluster.go

```diff
@@ -0,0 +1,248 @@
+		// Just wait for all syncers to be synced
+		for _, syncer := range syncers {
+			if !syncer.WaitUntilSynced(c.stop) {
+				log.Errorf("Timed out waiting for cluster %s to sync %v", c.ID, syncer)
+				continue
+			}
+		}
+		c.initialSync.Store(true)
+		return
+	}
+
+	// We're about to start modifying the cluster, so we need to lock it
+	if features.RemoteClusterTimeout > 0 {
+		time.AfterFunc(features.RemoteClusterTimeout, func() {
+			if !c.initialSync.Load() {
+				log.Errorf("remote cluster %s failed to sync after %v", c.ID, features.RemoteClusterTimeout)
+				timeouts.With(clusterLabel.Value(string(c.ID))).Increment()
+			}
+			c.initialSyncTimeout.Store(true)
+		})
+	}
+
+	opts := krt.NewOptionsBuilder(c.stop, fmt.Sprintf("ambient/cluster[%s]", c.ID), debugger)
+	opts = opts.WithMetadata(krt.Metadata{ClusterKRTMetadataKey: c.ID})
+	namespaces := kclient.New[*corev1.Namespace](c.Client)
+	// This will start a namespace informer and wait for it to be ready. So we must start it in a go routine to avoid blocking.
+	filter := filter.NewDiscoveryNamespacesFilter(namespaces, mesh, c.stop)
+	kube.SetObjectFilter(c.Client, filter)
+	// Register all of the informers before starting the client
+	defaultFilter := kclient.Filter{
+		ObjectFilter: c.Client.ObjectFilter(),
+	}
+
+	Namespaces := krt.WrapClient(namespaces, opts.WithName("informer/Namespaces")...)
+	Pods := krt.NewInformerFiltered[*corev1.Pod](c.Client, kclient.Filter{
+		ObjectFilter:    c.Client.ObjectFilter(),
+		ObjectTransform: kube.StripPodUnusedFields,
+	}, opts.WithName("informer/Pods")...)
+
+	gatewayClient := kclient.NewDelayedInformer[*v1beta1.Gateway](c.Client, gvr.KubernetesGateway, kubetypes.StandardInformer, defaultFilter)
+	Gateways := krt.WrapClient[*v1beta1.Gateway](gatewayClient, opts.WithName("informer/Gateways")...)
+	servicesClient := kclient.NewFiltered[*corev1.Service](c.Client, defaultFilter)
+	Services := krt.WrapClient[*corev1.Service](servicesClient, opts.WithName("informer/Services")...)
+
+	Nodes := krt.NewInformerFiltered[*corev1.Node](c.Client, kclient.Filter{
+		ObjectFilter:    c.Client.ObjectFilter(),
+		ObjectTransform: kube.StripNodeUnusedFields,
+	}, opts.WithName("informer/Nodes")...)
+
+	EndpointSlices := krt.NewInformerFiltered[*discovery.EndpointSlice](c.Client, kclient.Filter{
+		ObjectFilter: c.Client.ObjectFilter(),
+	}, opts.WithName("informer/EndpointSlices")...)
+
+	ConfigMaps := krt.NewInformerFiltered[*corev1.ConfigMap](c.Client, kclient.Filter{
+		ObjectFilter: c.Client.ObjectFilter(),
+	}, opts.WithName("informer/ConfigMaps")...)
+
+	if !c.Client.RunAndWait(c.stop) {
+		log.Warnf("remote cluster %s failed to sync", c.ID)
+		return
+	}
+
+	c.namespaces = Namespaces
+	c.gateways = Gateways
+	c.services = Services
+	c.nodes = Nodes
+	c.endpointSlices = EndpointSlices
+	c.pods = Pods
+	c.configmaps = ConfigMaps
+	c.initialized.Store(true)
+
+	// Reassign syncers for the check
+	syncers = []krt.Syncer{
```

Comment:
why do we stop checking configmaps here?

Reply:
Good catch; fixed